analyzers/src/SonarAnalyzer.CSharp/Extensions/TypeDeclarationSyntaxExtensions.cs

```diff
@@ -33,14 +33,19 @@ public static class TypeDeclarationSyntaxExtensions
         /// Returns a union of all the methods and local functions from a given type declaration.
         /// </summary>
         public static IEnumerable<IMethodDeclaration> GetMethodDeclarations(this TypeDeclarationSyntax typeDeclaration) =>
-            typeDeclaration
-                .Members
-                .OfType<MethodDeclarationSyntax>()
-                .SelectMany(method => GetLocalFunctions(method).Union(new List<IMethodDeclaration> { MethodDeclarationFactory.Create(method) }));
+            typeDeclaration.Members
+                           .OfType<MethodDeclarationSyntax>()
+                           .SelectMany(method => GetLocalFunctions(method).Union(new List<IMethodDeclaration> { MethodDeclarationFactory.Create(method) }));
 
-        private static IEnumerable<IMethodDeclaration> GetLocalFunctions(MethodDeclarationSyntax methodDeclaration)
-            => methodDeclaration.DescendantNodes()
-                .Where(member => member.IsKind(SyntaxKindEx.LocalFunctionStatement))
-                .Select(member => MethodDeclarationFactory.Create(member));
+        public static IEnumerable<IMethodDeclaration> GetMethodDeclarations(this CompilationUnitSyntax compilationUnitSyntax) =>
```

Comment:
This extension method should be placed in the corresponding file

Reply:
Indeed.
I'll wait for this [PR](https://github.com/SonarSource/sonar-dotnet/pull/5105/files) to be merged and I'll apply this change.